parseAndPopulate/populate.py

```diff
@@ -131,176 +131,196 @@ def __init__(self):
+                self.process_reload_cache.join()
+            else:
+                self._run_complicated_algorithms()
+            self.logger.info(
+                (
+                    f'Populate took {int(time.time() - self.start_time)} seconds with the main and '
+                    f'{"without" if self.args.simple else "with"} complicated algorithm'
+                )
+            )
+            # Keep new hashes only if the ConfD was patched successfully
+            if not self.errors:
+                self._update_files_hashes()
+        self.logger.info('Populate script finished successfully')
 
+    def _send_notification_about_running_script_by_api(self):
+        args_dict = dict(self.args.__dict__)
+        # Hide password from credentials argument
+        args_dict['credentials'] = args_dict['credentials'][0]
+        self.message_factory.send_populate_script_triggered_by_api(args_dict.items())
 
-def main(
-    scriptConf: BaseScriptConfig = ScriptConfig(),
-    message_factory: t.Optional[MessageFactory] = None,
-):
-    args = scriptConf.args
-    if args.api:
-        message_factory = message_factory or MessageFactory()
-        send_notification_about_running_script_by_api(args, message_factory)
-    log_directory = scriptConf.log_directory
-    yang_models = scriptConf.yang_models
-    temp_dir = scriptConf.temp_dir
-    cache_dir = scriptConf.cache_dir
-    json_ytree = scriptConf.json_ytree
-    yangcatalog_api_prefix = scriptConf.yangcatalog_api_prefix
-    global LOGGER
-    LOGGER = log.get_logger('populate', f'{log_directory}/parseAndPopulate.log')
-
-    confd_service = ConfdService()
-    redis_connection = RedisConnection()
-    LOGGER.info('Starting the populate script')
-    start = time.time()
-    if args.api:
-        json_dir = args.dir
-    else:
-        json_dir = create_dir_name(temp_dir)
-        os.makedirs(json_dir)
-    LOGGER.info('Calling parse_directory script')
-    try:
-        parse_directory = import_module('parseAndPopulate.parse_directory')
-        script_conf = configure_parse_directory(parse_directory, args, json_dir)
-        parse_directory.main(scriptConf=script_conf)
-    except Exception as e:
-        LOGGER.exception(f'parse_directory error:\n{e}')
-        raise e
+    def _initialize_json_dir(self):
+        if self.args.api:
+            self.json_dir = self.args.dir
+        else:
+            self.json_dir = os.path.join(self.temp_dir, uuid.uuid4().hex)
+            os.makedirs(self.json_dir, exist_ok=True)
 
-    body_to_send = {}
-    if args.notify_indexing:
-        body_to_send = prepare_for_es_indexing(
-            yangcatalog_api_prefix, os.path.join(json_dir, 'prepare.json'),
-            LOGGER, args.save_file_dir, force_indexing=args.force_indexing
-        )
+    def _run_parse_directory_script(self):
+        self.logger.info('Calling parse_directory script')
+        try:
+            parse_directory = import_module('parseAndPopulate.parse_directory')
```

Comment:
It works fine running as a script, but I couldn't debug it while triggering by an API call, because can't find a way to do it (I believe it works fine as well), If you know and can tell me how to connect a debugger to a running docker-compose, I'll appreciate your help